ext/drreg/drreg.c

```diff
@@ -402,7 +402,11 @@ drreg_event_bb_insert_late(void *drcontext, void *tag, instrlist_t *bb, instr_t
         /* Restore aflags to app value */
         LOG(drcontext, LOG_ALL, 3, "%s @%d."PFX" aflags=0x%x: lazily restoring aflags\n",
             __FUNCTION__, pt->live_idx, instr_get_app_pc(inst), aflags);
-        res = drreg_restore_aflags(drcontext, bb, inst, pt, false/*keep slot*/);
+        /* On AArch64, we cannot keep the slot, because spill_reg is used to save the
+         * aflags and this functions requires the slot to be released.
+         */
+        res = drreg_restore_aflags(drcontext, bb, inst, pt,
+                                   IF_AARCHXX_ELSE(true, false/*keep slot*/));
```

Comment:
I'm not 100% sure about this: this code can be subtle.  I would think that Dr. Memory's pattern code would have hit a problem on ARM if this were incorrect before.  This is saying whether or not to give up the TLS slot that's holding the app's aflags value.  Can you elaborate on your comment above: I don't follow.  Here we just want to bring the app's aflags back into the real nzcv.  Why does that need to be coupled with giving up our TLS slot?